app/controllers/stages_controller.rb

```diff
@@ -142,6 +141,9 @@ def stage_permitted_params
         deploy_group_ids: [],
         command_ids: []
       }
-    ] + Samson::Hooks.fire(:stage_permitted_params).flatten
+    ].tap do |params|
+      params << :no_code_deployed if current_user.admin?
```

Comment:
would actually be nice to do `alias_method :can_change_no_code_deployed?, :admin?` in User and then use `can_change_no_code_deployed?` everywhere so the consistency between view and controller is obvious
... alternatively do `helper_method def can_change_no_code_deployed?` in the controller and reuse in the view